### ISSUE TITLE
VOD gets successfully downloaded but also gets corrupted at the same time.

### ISSUE THREAD

**Issue by DieHardjagged** [lay295/TwitchDownloader#416]
### Checklist

- [X] I have checked the [issue page](https://github.com/lay295/TwitchDownloader/issues) for duplicates
- [X] I am running the latest version ([download here](https://github.com/lay295/TwitchDownloader/releases/latest))

### Describe your issue here

This issue occured with a download of a VOD that i had done previously, i retried the download 2 additional times to see if it was just an accident, but both times the individual files were marked as "completed" but still remained unusable, i noticed it immediately when the file that i got had no timelength marker like my previous downloads from before October 30th.
The downloaded chat along with said VOD came out fine and plays without problems.
This was not an issue for me when i downloaded another VOD previously on October 30th.

**Comment by ScrubN** [comment]
What does "marked as completed" mean?

**Comment by DieHardjagged** [comment]
> What does "marked as completed" mean?

Meaning that it was downloaded successfully, but the video file that i got is unusable.

**Comment by DieHardjagged** [comment]
Had no problems with previous VOD downloads, but then again, i used Version 1.50.5 of the Downloader for those.
I dont have the corrupted file anymore, but i will download it now, since its so easy to reproduce this problem.
Also never used the program for other VOD downloads, im focused on downloading this one specific VOD until i get a working version of it.

**Comment by ScrubN** [comment]
> Had no problems with previous VOD downloads, but then again, i used Version 1.50.5 of the Downloader for those.

1.50.6 did not have had any VOD downloading changes, so your issue should technically also be a problem on 1.50.5

**Comment by DieHardjagged** [comment]
Alright, so i tried downloading the same VOD again and i got the "failure" status after it kept being stuck at 90%. 
![Screenshot 2022-11-07 205115](https://user-images.githubusercontent.com/62212036/200401974-29cb2aca-d9e2-4e5f-84d4-6899d66507b1.jpg)

**Comment by DieHardjagged** [comment]
> 

Which is strange now, given i never had a single problem with 1.50.5.

**Comment by ScrubN** [comment]
> Alright, so i tried downloading the same VOD again and i got the "failure" status after it kept being stuck at 90%.

We need to add even a brief stacktrace to that. Just a status indicator is so unclear.

**Comment by DieHardjagged** [comment]
I downloaded the same VOD with version 1.50.5 now and also only got a corrupted file out of it even though it was marked as completed.

**Comment by ScrubN** [comment]
> I downloaded the same VOD with version 1.50.5 now and also only got a corrupted file out of it even though it was marked as completed. As a test, i downloaded something entirely different, a short clip that someone created on my own Twitch channel, zero problems downloading and playing it.

Sounds like it could be specific to the VOD or VODs in general? Can you test with other VODs and if it turns out to be that VOD in particular then please post the ID

**Comment by DieHardjagged** [comment]
> Sounds like it could be specific to the VOD or VODs in general? Can you test with other VODs and if it turns out to be that VOD in particular then please post the ID

Since you suggested it, i will download a different VOD from the channel i got the corrupted VOD from, as a sidenote, the VOD that is giving me trouble has alot of muted segmentes and i mean alot.
I dont exactly know what you mean with ID, but i think it should be1644280471?

**Comment by ScrubN** [comment]
I was able to successfully download the VOD via the CLI. Testing with the GUI task queue now. Also downloading at 160p because I don't have the bandwidth to download 7gb of footage multiple times, so that may affect my ability to replicate.

**Comment by DieHardjagged** [comment]
> 

Oh, now that you mention it, i never tested to download it on a lower resolution to see if it works or not.

> The GUI task queue finished without fail. Either the 160p affected the results or the issue is specific to you. Could you try downloading the VOD at 160p to confirm if that has an effect?

I will try downloading it, after im done downloading this different vod from the same channel.

**Comment by DieHardjagged** [comment]
> I was able to download `1644280471` via the GUI and CLI at 1080 (best) resolution without corruption. Perhaps ffprobe can give some more information. If you have a succesful, but corrupt video, can you run `ffprobe /path/to/video.mp4` and provide the results.

I still have a successfull but corrupted video on my SSD Drive, but i have no knowledge of how to run command lines since im only using the GUI version.

**Comment by Zibbp** [comment]
No worries. If you're still planning on uploading the corrupt video we can download that and use ffprobe on it.

**Comment by DieHardjagged** [comment]
> No worries. If you're still planning on uploading the corrupt video we can download that and use ffprobe on it.

Yes, im currently uploading it to my Google Drive.

**Comment by Zibbp** [comment]
FFprobe states the following for the mp4.
```
[mov,mp4,m4a,3gp,3g2,mj2 @ 0000021290182940] moov atom not found
```
Upon a quick Google search it appears this most commonly happens when the process creating the mp4 unexpectedly ends and cannot create the atom metadata which is one of the last steps.

**Comment by ScrubN** [comment]
For an easy solution to `moov atom not found`, I found a youtube tutorial that should walk you through the process: https://youtu.be/K3nWw9aX5kw

**Comment by ScrubN** [comment]
Now the question is why does the transcoding quit before adding the the moov atom? Perhaps it's because something screwed up on Twitch's end but the program assumes it wasn't?

**Comment by ScrubN** [comment]
> I was able to download `1644280471` via the GUI and CLI at 1080 (best) resolution without corruption.

How did you download from the GUI? Did you use task enqueuing? It's possible it's linked to that

**Comment by Zibbp** [comment]
> How did you download from the GUI? Did you use task enqueuing? It's possible it's linked to that

Downloaded without the task queue, directly from the "VOD Downloader" tab.

**Comment by Zibbp** [comment]
Just tried downloading `1644280471` via the task queue and I was able to re-produce the corrupted mp4. FFprobe throws the same error: `[mov,mp4,m4a,3gp,3g2,mj2 @ 0000029816222680] moov atom not found` Something with queueing up vods is screwing it up.

**Comment by DieHardjagged** [comment]
> For an easy solution to `moov atom not found`, I found a youtube tutorial that should walk you through the process: https://youtu.be/K3nWw9aX5kw

I followed it exactly as shown, but in the end i only got an unusable file out of it and since i cant get a working file in 720p/1080p anymore, all i can do now is try to get an 480p version which means now i will have 11 or 12 days left to get a 1080p file because after that timeframe the past broadcast will be gone.

**Comment by Zibbp** [comment]
Until the issue is located, you should be able to download the 1080p VOD using the "VOD Downloader" tab, **not** the "Task Queue" tab. I did not have any issues downloading the VOD this way.
![image](https://user-images.githubusercontent.com/21207065/200625099-ab771b43-0b40-42ae-8572-6a441754c000.png)

**Comment by DieHardjagged** [comment]
![Screenshot 2022-11-08 165630](https://user-images.githubusercontent.com/62212036/200613470-f1098763-090b-45ea-a919-62b6b976690b.png)


> Until the issue is located, you should be able to download the 1080p VOD using the "VOD Downloader" tab, **not** the "Task Queue" tab. I did not have any issues downloading the VOD this way. ![image](https://user-images.githubusercontent.com/21207065/200625099-ab771b43-0b40-42ae-8572-6a441754c000.png)

I see, i never used the direct download option, because i always used the queue option since i also downloaded the Chat etc alongside it, thanks, i will try it that way now.

**Comment by ScrubN** [comment]
>and since i cant get a working file in 720p/1080p anymore

I downloaded `1644280471` at 720p60 using enqueue and enabled all logging on ffmpeg and it completed the transcode fine. There were about 510 corrupt packets but other than that nothing unusual happened. I can try again with 1080p60 but that'll take a few more hours.

**Comment by ScrubN** [comment]
The 1080p60 download finished identically to the 720p60 one. The transcode went fine and again had around 500 corrupt packets.

I don't know if the difference is because of my computer, or the fact I used only 2 download threads, or because I'm debugging on master, or because I setup breakpoints, but I'm unable to reproduce it in an environment where I can actually see what the program is doing.

Seeing as one of your downloads failed at 90% I'd have to assume either Twitch temporarily blocked your IP or your internet dropped out for more than 10 times.

<br>
Speaking of, it would probably be wise to change from checking total error counts to logging error times as on a slower or unstable connection it would not be unlikely to drop more than 10 times over the course of let's say 1-4 hours of downloading the same VOD. Then you can check the total error count during for example the last 30 minutes to better estimate if the download should be canceled.

**Comment by lay295** [comment]
I've tried many times with both the queue and the main download page to try and reproduce this and I can't. Tried every resolution on the VOD in both ways and couldn't get it.

If anybody does manage to reproduce it and gets all the .ts files before the program cleans it up that would be pretty awesome.

@ScrubN yeah... the VOD/Clip downloading of this program was kind of an after thought :) I usually just point people towards TwitchLeecher but well... can't do that anymore I suppose. Ideally I would want to handle errors a bit more intelligently on the VOD side (like scaling back concurrent download connections or something).

**Comment by DieHardjagged** [comment]
Alright, for the last several hours i tried to reproduce this error, but strangely during my two attempts so far, neither did i get a corrupted file, i downloaded VOD (ID 1644280471) twice, once in 720p resolution and then again in 1080p resolution and both times i got a working file, i do know that my Internet was not dropping during the times when i got the corrupted files, but i cant say for sure.

**Comment by Zibbp** [comment]
Downloaded `1644280471` via the queue again at 1080p and this time the VOD had no corruption or playback issues. Must have been something on Twitch's end?

**Comment by DieHardjagged** [comment]
> Downloaded `1644280471` via the queue again at 1080p and this time the VOD had no corruption or playback issues. Must have been something on Twitch's end?

It could very well be that the problem was caused on Twitchs end.

**Comment by ScrubN** [comment]
Yeah I'd be inclined to assume the issue was on Twitchs end, probably they were still processing it but it was very low priority because small streamer. Nonetheless I would still like to change the error counter like I mentioned.

**Comment by DieHardjagged** [comment]
Since this problem resolved by itself i am going to close the Topic, thanks again for all the help.
Cheers

**Comment by DieHardjagged** [comment]
Well, i am reopening this Thread because it happened again with VOD 1669299166 as in "Download failed" at 97% through the Queue option, guess i just have to download the VODs directly and not through the Queue option, thought this problem would be fixed with future updates of the program, but i guess it will happen either way, i was also monitoring my Internet connection and know that it never dropped.
Cheers

**Comment by ElBuenAnvita** [comment]
Experienced same issue twice this week using v1.50.7 [first time]/v1.51.0 [second time]. I usually Download VOD + Chat JSON via Task Queue. Both times the corruped videos were the last VOD added in the queue and both videos were downloaded/rendered AFTER a VOD that was downloaded with the "Crop" option.

This is a theory but maybe the crop option may be interferring with the next video in queue. Or maybe that it's the last one in queue. Gotta check that later when I continue to downloading another VODs. Or maybe everything it's just a coincidence.

This is my current shown Task Queue in TwitchDownloader:
![image](https://user-images.githubusercontent.com/37051987/207479879-1b8bf7fe-8fb5-41c5-8a83-979b3ce21a7b.png)

Here are the files from that task queue in the file explorer; the selected one is the corrupted one:
![image](https://user-images.githubusercontent.com/37051987/207480008-8fb7bcf7-5652-421b-ba3a-1380d542056d.png)

These are the details of the timeline of VODs downloaded + notes explaining (I always make a list of VODs I'll download):
![image](https://user-images.githubusercontent.com/37051987/207481065-344ae7a3-10f2-42bd-8016-3fbaddb6b16b.png)

Command `ffprobe` on corruped video also throws `moov atom not found`.
![image](https://user-images.githubusercontent.com/37051987/207481536-bb58d925-1045-4635-a330-4c0b60b73282.png)

I hope this information may be helpful.

> If anybody does manage to reproduce it and gets all the .ts files before the program cleans it up that would be pretty awesome.

And sorry, the temp folder was already cleaned and no .ts files I found there.

A sidenote: Before pushing this comment, I noticed I was running out of space in my computer [currently 4,85 GB availabe out of 476 GB], (maybe the encoding was mid-stopped because it run out of space...?)

I'm going to continue downloading and retrying download this VOD. I'll update over here if it happens again.
